src/ray/raylet/node_manager.cc

```diff
@@ -1157,6 +1157,11 @@ void NodeManager::HandleNotifyGCSRestart(const rpc::NotifyGCSRestartRequest &req
 
 void NodeManager::UpdateResourceUsage(const NodeID &node_id,
                                       const rpc::ResourcesData &resource_data) {
+  // Trigger local GC at the next heartbeat interval.
```

Comment:
Why move this logic at he very beginning of the method?

Reply:
The head node is not in the local `ClusterResourceManager`'s resource view. So if there is a global gc signal sent from  the head node, it will be blocked by the `if (!cluster_resource_scheduler_->GetClusterResourceManager().UpdateNode())` statement.